src/main/webapp/js/finscholar/commonlistview/commonlistview.soy

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{namespace finscholar.commonlistview.templates}
+
+{template .commonlistview}
+  {@param pageIndex : string}
+  <table class="mdl-data-table mdl-js-data-table 
+      mdl-data-table--selectable mdl-shadow--2dp">
+    <thead>
+      {if $pageIndex == '1'}
+        {call .scholarshiplistheader /}
+      {else}
+        {call .collegelistheader /}
+      {/if}
+    </thead>
+    <tbody id="table-body">
+    </tbody>
+  </table>
+{/template}
+
+{template .scholarshiplistheader}
+  <tr>
+    <th class="mdl-data-table__cell--non-numeric">Scholarship Name</th>
+    <th>Schools</th>
+    <th>Amount</th>
+  </tr>
+{/template}
+
+{template .collegelistheader}
+  <tr>
+    <th class="mdl-data-table__cell--non-numeric">College Name</th>
+    <th>Acceptance Rate</th>
+    <th>Average ACT</th>
+  </tr>
+{/template}
+
+{template .scholarshiplistitems}
+  {@param scholarships: list<[name: string, schools: string, amount: string]>}
+  {for $item in $scholarships}
+    <tr jsaction="scholarshiplistview.clickAction;
```

Comment:
Ah ok so you're binding a jsaction click event handler to each individual list item. I wonder how we'll transfer control to the single scholarship view and single college view whenever it gets clicked? In other words, how do we persist data from the item in the list of scholarships? Will we just pass in the item and process its data in the single views?

Reply:
The implementation for that part is in #48 . The code is in homePageController